### PR TITLE
OCC-238: GitHub Actions Tests fails for Windows

### LINKED ISSUE #433: GitHub Actions Tests fails for Windows (OCC-238)
### Describe the bug

Build and Test (Windows) fails with a timeout for the main branch. 
https://github.com/OrchardCMS/OrchardCore.Commerce/actions/runs/8804049904/job/24378106668

[Jira issue](https://lombiq.atlassian.net/browse/OCC-238)

### PR DESCRIPTION
**Pull request by sarahelsaig**
[OCC-238](https://lombiq.atlassian.net/browse/OCC-238)
Fixes #433

### CODE REVIEW

**Comment by sarahelsaig** [comment]
I did 3 things in this PR:
- Update UITT to the latest alpha (see https://github.com/Lombiq/UI-Testing-Toolbox/issues/358#issuecomment-2083987263)
- Change _.htmlvalidate.json_ to _orchardcore.htmlvalidate.json_ to avoid local Windows tests randomly disregarding the `"form-dup-name": "off"` because they were loading in the _.htmlvalidate.json_ from UITT instead of the one in the OrchardCore.Commerce.Tests.UI project.
- Added per-test timeout so it's easier to verify when individual tests are hanging.

**Review by Piedone — CHANGES_REQUESTED** [review]

File: Directory.Packages.props [hunk 1]
```diff
@@ -9,9 +9,9 @@
     <PackageVersion Include="Lombiq.HelpfulLibraries.Refit" Version="9.1.1-alpha.6.occ-203" />
     <PackageVersion Include="Lombiq.NodeJs.Extensions" Version="2.1.0" />
     <PackageVersion Include="Lombiq.Tests" Version="3.0.0" />
-    <PackageVersion Include="Lombiq.Tests.UI" Version="9.0.1-alpha.2.occ-218" />
-    <PackageVersion Include="Lombiq.Tests.UI.AppExtensions" Version="9.0.1-alpha.2.occ-218" />
-    <PackageVersion Include="Lombiq.Tests.UI.Shortcuts" Version="9.0.1-alpha.2.occ-218" />
+    <PackageVersion Include="Lombiq.Tests.UI" Version="9.0.1-alpha.7.osoe-840" />
+    <PackageVersion Include="Lombiq.Tests.UI.AppExtensions" Version="9.0.1-alpha.7.osoe-840" />
+    <PackageVersion Include="Lombiq.Tests.UI.Shortcuts" Version="9.0.1-alpha.7.osoe-840" />
```
[the remark this answers is not in the document]
Comment (sarahelsaig) [reply Directory.Packages.props:14]:
```suggestion
    <PackageVersion Include="Lombiq.Tests.UI" Version="9.1.0" />
    <PackageVersion Include="Lombiq.Tests.UI.AppExtensions" Version="9.1.0" />
    <PackageVersion Include="Lombiq.Tests.UI.Shortcuts" Version="9.1.0" />
```

File: test/OrchardCore.Commerce.Tests.UI/UITestBase.cs [hunk 2]
```diff
@@ -16,7 +16,26 @@ protected UITestBase(ITestOutputHelper testOutputHelper)
...
+        var testTask = ExecuteTestAsync(testAsync, browser, SetupHelpers.RunSetupAsync, changeConfigurationAsync);
+        var timeoutTask = Task.Delay(timeoutValue);
+
+        await Task.WhenAny(testTask, timeoutTask);
+
+        if (!testTask.IsCompleted)
+        {
+            throw new TimeoutException($"The time allotted for the test ({timeoutValue}) was exceeded.");
+        }
+    }
```
[the remark this answers is not in the document]
Comment (sarahelsaig) [reply test/OrchardCore.Commerce.Tests.UI/UITestBase.cs:38]: I will backport it to UITT in https://github.com/Lombiq/UI-Testing-Toolbox/issues/359

File: test/OrchardCore.Commerce.Tests.UI/UITestBase.cs [hunk 3]
```diff
@@ -45,6 +64,11 @@ protected UITestBase(ITestOutputHelper testOutputHelper)
                     OrchardCoreUITestExecutorConfiguration.AssertBrowserLogIsEmpty(messageWithoutJqueryError);
                 };
 
+                configuration.HtmlValidationConfiguration.HtmlValidationOptions =
+                    configuration.HtmlValidationConfiguration.HtmlValidationOptions
+                        .CloneWith(validationOptions => validationOptions.ConfigPath =
+                            Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "orchardcore.htmlvalidate.json")); // #spell-check-ignore-line
```
[the remark this answers is not in the document]
Comment (sarahelsaig) [reply test/OrchardCore.Commerce.Tests.UI/UITestBase.cs:70]: Sure, but the effort has already been extended here, so it's an okay stop-gap in the meantime.

**Pull request merged**